pkg/azure/api/api.go

```diff
@@ -351,6 +351,8 @@ func parseSubnet(subnet *network.Subnet) (s *ipamTypes.Subnet) {
 		s.CIDR = c
 		if subnet.IPConfigurations != nil {
 			s.AvailableAddresses = c.AvailableIPs() - len(*subnet.IPConfigurations)
+		} else {
+			s.AvailableAddresses = c.AvailableIPs()
```

Comment:
Looking at the Azure NRP code and logs, I believe `IPConfigurations` can be set to "nil" when the number of ipconfigs exceeds a limit. I'm following up internally to see if anything changed here.

Reply:
re-posting from slack convo : 
In this case, we aren't seeing the expected `IpConfigurationsNextLink` in the response either unfortunately.

Reply:
confirmed with NRP team that `IPConfigurations` is nil occurs when the number of configurations exceeds a limit. At that point NRP returns a "paginated" response, which is supposed to contain `IpConfigurationsNextLink`. However, `IpConfigurationsNextLink` is not currently in the API spec, so the ARM API clients don't see it. Waiting on confirmation, but I believe this is a bug in the API spec.